ext/pdo/pdo_stmt.c

```diff
@@ -1786,7 +1786,8 @@ bool pdo_stmt_setup_fetch_mode(pdo_stmt_t *stmt, zend_long mode, uint32_t mode_a
 				return false;
 			}
 
-			ZVAL_COPY(&stmt->fetch.into, &args[0]);
+			GC_TRY_ADDREF(Z_OBJ(args[0]));
```

Comment:
Why try?

Reply:
I don't know why I did this, I think I didn't want to get bit by SHM stuff, forgetting that objects can't live in it. (although can enums live in SHM?)

Reply:
Objects indeed can't live inside shm, and so enums can't either. If you use a default value with an enum you're actually storing an IS_CONSTANT_AST and deferring the evaluation into an object to the runtime.